debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresSchemaIT.java

```diff
@@ -236,15 +238,46 @@ public void shouldDetectNewChangesAfterRefreshing() throws Exception {
 
         TestHelper.execute(statements);
         try (PostgresConnection connection = TestHelper.create()) {
-            schema.refresh(connection, TableId.parse(tableId, false));
+            schema.refresh(connection, TableId.parse(tableId, false), false);
             assertTablesIncluded(tableId);
             assertTablesExcluded("public.table1");
             assertTableSchema(tableId, "vc, si",
-                              Schema.OPTIONAL_STRING_SCHEMA, Schema.OPTIONAL_INT16_SCHEMA);
+                    Schema.OPTIONAL_STRING_SCHEMA, Schema.OPTIONAL_INT16_SCHEMA);
             assertColumnsExcluded(tableId + ".strcol");
         }
     }
 
+    @Test
+    public void shouldPopulateToastableColumnsCache() throws Exception {
+        String statements = "CREATE SCHEMA IF NOT EXISTS public;" +
+                "DROP TABLE IF EXISTS table1;" +
+                "CREATE TABLE table1 (pk SERIAL,  toasted text, untoasted int, PRIMARY KEY(pk));";
+        TestHelper.execute(statements);
+        PostgresConnectorConfig config = new PostgresConnectorConfig(TestHelper.defaultConfig().build());
+        schema = TestHelper.getSchema(config);
+        TableId tableId = TableId.parse("public.table1", false);
+
+        // Before refreshing, we should have an empty array for the table
+        assertTrue(schema.getToastableColumnsForTableId(tableId).isEmpty());
+
+        try (PostgresConnection connection = TestHelper.create()) {
+            // Load up initial schema info. This should not populate the toastable columns cache, as the cache is loaded
+            // on-demand per-table refresh.
+            schema.refresh(connection, false);
+            assertTrue(schema.getToastableColumnsForTableId(tableId).isEmpty());
+
+            // After refreshing w/ toastable column refresh disabled, we should still have an empty array
+            schema.refresh(connection, tableId, false);
+            assertTrue(schema.getToastableColumnsForTableId(tableId).isEmpty());
+
+            // After refreshing w/ toastable column refresh enabled, we should have only the 'toasted' column in the cache
+            schema.refresh(connection, tableId, true);
+            List<String> toastableColumns = schema.getToastableColumnsForTableId(tableId);
```

Comment:
Super-minor nit: this should be done with AssertJ like so (which we already use in the project):

    assertThat(schema.getToastableColumnsForTableId(tableId)).containsOnly("toasted");

Putting stylistic preferences aside, it's advantageous if the assertion on size below fails. In this case you'll be interested in the actual contents of the list, which is revealed in the message if `containsOnly()` fails.

No need for you to change it, I'll do it when integrating this.